workshop/nano1-vuex.md

```diff
@@ -0,0 +1,281 @@
+    
+    context.commit("addGame", game);
+  }
+}
+```
+
+In our action, we are making a micro validation for the data we will insert. This is just an example
+of what you can do inside this function, you could also make axios/ajax calls and commit data on the
+responses!
+
+`context.commit` will call our *addGame* **mutation** and pass the data after we validate it.
+
+Inside our mutation, we simply push the game into our array.
+
+## 4. Step 3 - Getting our data from the store
+If you run the application at this point nothing will happen, but our data has already been 
+added to our vuex store.
+
+In order to use our Vue development tools, first click on the `Open in new window` button on the
+right side toolbar (above the white canvas area). This will open a new browser tab where
+you can navigate to your Vue devtools through the development console.
+
+Open Vue, and click on the clock icon to switch to the Vuex view.
+
+You will see that you have your `state` object with our array of games inside of it!
+
+Head back to our code, and inside `App.vue` add a new **computed property**:
+
+```js
+games() {
+  return this.$store.state.games;
+}
+```
+
+Here, we are going inside our `$store` and fetching the *games* array from the **state**.
+
+The cards will now populate with all our store data :)
+
+Another way to fetch data from the store is by using **getters**.
+
+Go to our `store.js` file and add the following getter:
+```js
+gameTitles(state) {
+  return state.games.map(game => game.title);
+}
+```
+
+You will notice that the difference between using a getter and accesing our state directly like above,
+is that here we have a chance to parse or modify our data before sending it back to the caller. In
+this case, were extracting an array of only the game titles. Lets output this on our app.
+
+Add this computed property to `App.vue`:
+
+```js
+gameTitles() {
+  return this.$store.getters.gameTitles;
+}
+```
+
+Our `<ul>` element will now be populated with the titles of our games!
+
+## 5. Step 4 - Modifying our data
+At this point you have all the tools you need to be able to start using Vuex in your apps, but lets make one more 
+change and give those **Delete** buttons some functionality.
+
+Go to `store.js` and add this **action**:
+
+```js
+removeGame({ commit, state }, game) {
+  const games = [...state.games];
+  const storedGame = games.find(g => g.title === game.title);
+  if (!storedGame) return;
+
+  games.splice(games.indexOf(storedGame), 1);
+  commit("setGames", games);
+}
+```
+
+Here we grab a copy of our `state.games` array into a `games` constant.
+(It's important that we make a copy, since we should **NEVER** modify our state directly!).
+
+We find the game that we want to delete by its title, and we commit a new mutation by the name of `setGames`
+which will allow us to modify the whole array on one go.
+
+::: tip 💡
+Notice that we are using ES6 object destructuring in the first param. This way instead of having to pass
```

Comment:
Let's add a link to destructuring
```suggestion
Notice that we are using [ES6 object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring) in the first param. This way instead of having to pass
```